src/components/CountriesList.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Link } from 'react-router-dom'
+
+const CountriesList = props => {
+    const allCountries = props.countries.map(country => (
+
+        <li className="noDecoration">
+            <Link to={`/country/${country.cca3}`} className="list-group-item list-group-item-action">
+                <img src={`https://www.countryflags.io/${country.cca2}/flat/64.png`} className="flag" alt="flag"/>
+                {`${country.name.common}`}
+            </Link>
+        </li>
+    ))
```

Comment:
```suggestion
    const allCountries = props.countries.map((country, idx) => (

        <li key={idx} className="noDecoration">
            <Link to={`/country/${country.cca3}`} className="list-group-item list-group-item-action">
                <img src={`https://www.countryflags.io/${country.cca2}/flat/64.png`} className="flag" alt="flag"/>
                {`${country.name.common}`}
            </Link>
        </li>
    ))
```
Las keys ayudan a React a identificar que ítems han cambiado, son agregados, o son eliminados. Las keys deben ser dadas a los elementos dentro del array para darle a los elementos una identidad estable.

Cuando no tengas IDs estables para renderizar, puedes usar el índice del ítem como una key como último recurso.

No es recomendable usar índices para keys si el orden de los ítems puede cambiar. Esto puede impactar negativamente el rendimiento y puede causar problemas con el estado del componente.